lib/atlas/dataset.rb

```diff
@@ -114,6 +114,11 @@ class Dataset
 
     validates :interconnector_capacity, numericality: true
 
+    # Public: Check if the dataset is derived from another dataset
+    def derived?
```

Comment:
This way the `Dataset` knows about the existence of derived datasets. It should not have to know anything about that.

What I meant was a helper method inside Atlas::Runner:
```
def precomputed_graph?
  dataset.is_a?(Dataset::DerivedDataset)
end
```